universe/syncer.go

```diff
@@ -338,6 +349,17 @@ func (s *SimpleSyncer) syncRoot(ctx context.Context, remoteRoot Root,
 		return err
 	}
 
+	// We use an error group to simply the error handling of a goroutine.
+	// This goroutine will handle reading in batches of new leaves to
+	// insert into the DB. We'll fee the output of the goroutines below
+	// into the input fetchedLeaves channel.
+	batchSyncEG.Go(func() error {
+		newLeafProofs, err = s.batchStreamNewItems(
```

Comment:
Itest failure is caused by the leaves being inserted above not being added to `newLeafProofs`, which is used to build the RPC response.

I verified a fix locally by declaring `newAnchorProofs`, populating that as suggested above, and then joining the two lists before sending the `AssetSyncDiff`:

```go
newLeafProofs = append(newAnchorProofs, newLeafProofs...)
```